ompi/mca/op/avx/op_avx_functions.c

```diff
@@ -77,22 +75,20 @@
     int size = *count/step; \
     int i; \
     int round = size*64; \
-    type* in = (type*)_in; \
-    type* out = (type*)_out; \
     for (i = 0; i < round; i+=64) { \
-        __m512i vecA =  _mm512_loadu_si512((in+i));\
-        __m512i vecB =  _mm512_loadu_si512((out+i));\
+        __m512i vecA =  _mm512_loadu_si512((_in+i));\
```

Comment:
Definition for _mm512_load_si512: 
 __m512i _mm512_load_si512 (void const* mem_addr)

I am not sure we need to do type convert. The types_per_step will be different based on the type, I tried to calculate the correct step for each type, but it seems not working. 

For correctness purpose I will remove type convert now. If we really need to specific type. I will fix later.

Reply:
This is plain ugly and highly non portable. I suggest you put back the typecast and add not 64 to i but ```step```. In fact you should model this on the float version below, without using the mask for the last set of operations but instead falling back to a lower vector op and then a Duff device.